src/app/tests/TestCommandInteraction.cpp

```diff
@@ -1053,6 +1152,62 @@ void TestCommandInteraction::TestCommandHandlerInvalidMessageSync(nlTestSuite *
     NL_TEST_ASSERT(apSuite, ctx.GetExchangeManager().GetNumActiveExchanges() == 0);
 }
 
+void TestCommandInteraction::TestCommandReplyLimits(nlTestSuite * apSuite, void * apContext)
+{
+    TestContext & ctx = *static_cast<TestContext *>(apContext);
+
+    // This test validates boundary conditions on replies
+    //   - replies will pack a single octet string of the specified size
+    //   - there will be some overhead for command paths and packing, hence
+    //     the min size range
+    //
+    // At the time of writing this, kMaxSecureSduLengthBytes is 1194
+    constexpr uint32_t kMinSize = chip::app::kMaxSecureSduLengthBytes - 100;
+    constexpr uint32_t kMaxSize = chip::app::kMaxSecureSduLengthBytes;
+
+    // these asserts is to understand the code ranges below ARE hit
+    static_assert(kMinSize < 1100);
+    static_assert(kMaxSize > 1180);
+
+    for (uint32_t replySize = kMinSize; replySize <= kMaxSize; replySize++)
+    {
+        if ((replySize >= 1145) && (replySize <= 1161)) {
+            continue; // Unclear why these fail
+        }
+
+        if ((replySize >= 1162) && (replySize <= 1165)) {
```

Comment:
If I understand this correctly it has to do with the Rollback/Checkpoint stuff. We fail in just the right way that we rollback to a state where the array closed, but the InvokeReponseMessage fails and then it proceeds to rollback

Reply:
This is what the test intends to validate - I am surprised we have 2 failure modes though. For very large buffers we obviously eventually succeed.

Reply:
Chatted about this with Andrei. But will post here for folks reading.

The second issue `Crash due to MessageBuilder::EncodeInteractionModelRevision() encoding failure` is likely because it error on `ReturnErrorOnFailure(mInvokeResponseBuilder.EndOfInvokeResponseMessage());` [here](https://github.com/project-chip/connectedhomeip/blob/b5dfe72508a27018af00520ac895f8e24c002385/src/app/CommandHandler.cpp#L542) while in `TryAddResponseData`, this cause a rollback where `mInvokeResponseBuilder.GetInvokeResponses().` is in the wrong state and no longer has an outer container, when it gets call a second time in the `FinishStatus` you get the error.

The first error is likely some other weirdness of rollback interacting in a weird way where we end some of the containers

Reply:
It looks like OpenIOT and NRF have different limits here (guessing maybe LWIP buffer allocation ranges maybe).

I am unsure if we want to try to have this test in and then improve or just use it as a template to fix the bugs we have and then commit some version of it only once it consistently passes ...